src/components/ProjectSettingsTabs/NotificationsTab/types.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  buildRegexSelectors,
+  taskRegexSelectors,
+  requesterSubscriberConfig,
+  ResourceType,
+  Trigger,
+  failureTypeSubscriberConfig,
+} from "constants/triggers";
+import { validateDuration, validatePercentage } from "utils/validators";
+
+export const triggers: Trigger[] = [
+  {
+    trigger: "outcome",
+    resourceType: ResourceType.VERSION,
+    label: "Any version finishes",
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: "failure",
+    resourceType: ResourceType.VERSION,
+    label: "Any version fails",
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: "outcome",
```

Comment:
Can the triggers also be defined as constants (as well as the extra field Keys)? I think this would help make it clearer that evergreen relies on these fields being precise to build the notification (as opposed to label, for example, which is strictly a UI element). I realize that this isn't done in the legacy UI but I don't think that defining these as constants there was the right thing to do either.

An alternative would be to get this list of triggers from Evergreen/globals.go or notifications.go with a query, but I recognize that might be an unnecessary slowdown.

Reply:
good point, done.